plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -87,6 +89,11 @@ private ConnectorSplit toIcebergSplit(FileScanTask task)
         //       The predicate here is used by readers for predicate push down at reader level,
         //       so when we do not use residual expression, we are just wasting CPU cycles
         //       on reader side evaluating a condition that we know will always be true.
+        if (!task.deletes().isEmpty()) {
```

Comment:
IMO this should be the only change in this PR.

Disallowing v2 tables even if there are no row-level deletes applied sounds a bit extreme.

Reply:
Yeah - I agree. I would very much prefer to only fail tables with delete markers.

Reply:
I guess this is more tricky to write test though. Does spark already support row-level deletes? If so we can use PT env we already have I think.

Reply:
I guess my thought is that with the spec not being final there's a chance other breaking changes will get added that this change won't catch. I don't know if there's still active development going on with the spec though.